site/_posts/2021-05-11-manageiq-perf-results.md

```diff
@@ -0,0 +1,86 @@
+
+### Steady state CPU utilization
+<img src="/assets/images/blog/PodSteadyStateCPU.png" width="250"></a>
+![](/assets/images/blog/PodSteadyStateCPU.png)
+
+The figure above shows the average process/pod CPU usage over a 2 hour period with the VMware provider running.  The appliance results (in blue) show the process-level CPU utilization based on data from the Linux ps command.   The podified results (in orange) show the pod-level CPU utilization from prometheus/grafana.  For most worker pods, CPU utilization in podified mode is consistently 8-9% higher than appliance mode.  In investigating this difference, the development team found that the extra overhead was caused by the liveness probe.  [Rewriting and optimizing the liveness check](https://github.com/ManageIQ/manageiq/pull/688) resulted in a 20x speedup.  The podified results with the optimized liveness check show up as gray bars in the figure.  The worker pods each show an 8-9% reduction in steady state CPU utilization.  This performance improvement are included in the Lasker branch. 
+
+The figure above also shows that the orchestrator CPU usage is lower in podified mode.  This is because the orchestrator pod doesn’t have to do the extra work of collecting worker CPU and memory usage, tracking worker heartbeats, and managing resource limits as it does in appliance mode.  For podified mode, this extra work is handled by Kubernetes.
+
+ <img src="/assets/images/blog/TotalSteadyStateCPU.png" width="250"></a>
+![](/assets/images/blog/TotalSteadyStateCPU.png)
+
+The figure above shows the total steady state ManageIQ CPU usage across all processes/pods for three workload scenarios:
+* No providers defined
+* 1 VMware provider
+* 13 providers
+For each workload scenario, the podified CPU usage is dramatically reduced (ranging from 74 to 94% lower) with the liveness probe improvement.  With the improvement, total ManageIQ podified CPU usage is lower than appliance mode in each case.
+
+### VMDB database measurements
+The ManageIQ architecture includes the Virtual Management Database (VMDB), which is "the definitive source of intelligence collected about your Virtual Infrastructure".  During our performance measurements, we collected postgres statistics to see the total size, number of rows and activity for each of the 300+ tables in the VMDB database.
+
+The biggest tables in the vmdb database are related to metric collection.  The following observations are based on the postgres stats and discussion with development:
+* 24 raw tables, one per hour (metrics_00, metrics_01 … metrics_23)
+  * Only the most recent four hours of raw metrics are kept, and every hour, the table with data older than 4 hours is dropped.
+  * For each resource, metrics are collected every 20 seconds.  Each resource gets one metric row per collection interval, with up to 180 rows collected per hour.  

```

Comment:
```suggestion
  * For each resource, metrics are collected in 20 second intervals.  Each resource gets one metric row per collection interval, with up to 180 rows collected per hour.  

```

Otherwise it sounds like we go out an collect it from the provider every 20 seconds, which isn't true.